flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -236,6 +236,89 @@ SqlDrop SqlDropTable(Span s, boolean replace) :
     }
 }
 
+/**
+* Parses an INSERT statement.
+*/
+SqlNode RichSqlInsert() :
+{
+    final List<SqlLiteral> keywords = new ArrayList<SqlLiteral>();
+    final SqlNodeList keywordList;
+    SqlNode table;
+    SqlNodeList extendList = null;
+    SqlNode source;
+    final SqlNodeList partitionList = new SqlNodeList(getPos());
+    SqlNodeList columnList = null;
+    final Span s;
+}
+{
+    (
+        <INSERT>
+    |
+        <UPSERT> { keywords.add(SqlInsertKeyword.UPSERT.symbol(getPos())); }
+    )
+    { s = span(); }
+    SqlInsertKeywords(keywords) {
+        keywordList = new SqlNodeList(keywords, s.addAll(keywords).pos());
+    }
+    <INTO> table = CompoundIdentifier()
+    [
+        LOOKAHEAD(5)
+        [ <EXTEND> ]
+        extendList = ExtendList() {
+            table = extend(table, extendList);
+        }
+    ]
+    [
+        LOOKAHEAD(2)
+        { final Pair<SqlNodeList, SqlNodeList> p; }
+        p = ParenthesizedCompoundIdentifierList() {
+            if (p.right.size() > 0) {
+                table = extend(table, p.right);
+            }
+            if (p.left.size() > 0) {
+                columnList = p.left;
+            }
+        }
+    ]
+    [
+        <PARTITION> PartitionSpecCommaList(partitionList) {
+            if (this.conformance != FlinkSqlConformance.HIVE) {
+                throw new ParseException("PARTITION expression encountered in illegal context");
```

Comment:
Need a cleaner error message

Reply:
Changed to `PARTITION expression is only allowed for HIVE dialect`

Reply:
INSERT INTO PARTITION is not allowed in current ${this.conformance} dialect ?

Reply:
insert into partition is supported as a HIVE dialect now, same as insert overwrite